### ISSUE TITLE
server side external io api logic broken

### ISSUE THREAD

**Issue by sss123next** [FreeRDP/FreeRDP#6664]
currently impossible to override `io.TransportAttach` because `->context->rdp->transport` is available only after `freerdp_peer_context_new` call, but inside of `freerdp_peer_context_new_patched` `transport_attach` is called before any chance to override it, it is possible to create own implementation of `freerdp_peer_context_new`, but it will also add requirement for implementation of lot freerdp internals which is undesired and should be avoided if possible.

**Comment by sss123next** [comment]
also looks like some  in some other places io descriptor used directly in server side (peer) code

**Comment by sss123next** [comment]
looks like another task for me (.
ok, i am on it.

**Comment by sss123next** [comment]
https://github.com/FreeRDP/FreeRDP/pull/6668 - started work here
